apps/backend/db_patches/0158_AddFAPReviewTemplate.sql

```diff
@@ -0,0 +1,68 @@
+DO
+$$
+DECLARE
+    fap_review_template_id_var int;
+    fap_review_topic_id_var int;
+    questionary_id_var int;
+BEGIN
+    IF register_patch('AddFAPReviewTemplate.sql', 'Gergely Nyiri', 'Add FAP review template', '2024-06-11') THEN
+
+    INSERT INTO template_categories(template_category_id, name) VALUES(10, 'FAP Review');
+	
+    INSERT INTO template_groups (template_group_id, category_id) VALUES('FAP_REVIEW_TEMPLATE', 10);
+	
+    INSERT INTO question_datatypes(question_datatype_id) VALUES('FAP_REVIEW_BASIS');
+
+    INSERT INTO questions(
+                question_id,
+                data_type,
+                question,
+                default_config,
+                natural_key,
+                category_id
+            )
+        VALUES(
+                'fap_review_basis'
+                , 'FAP_REVIEW_BASIS'
+                , 'FAP review basic information'
+                , '{"required":false,"small_label":"","tooltip":""}'
+                , 'fap_review_basis'
+                , 10
+            );
+
+    ALTER TABLE fap_reviews ADD COLUMN questionary_id INTEGER REFERENCES questionaries (questionary_id);
+
+    ALTER TABLE call ADD COLUMN fap_review_template_id INTEGER REFERENCES templates(template_id);
+
+    INSERT INTO templates(name, description, is_archived, group_id) VALUES 
+	('default fap review template', 'default fap review template', false, 'FAP_REVIEW_TEMPLATE');
+
+    SELECT templates.template_id 
+      INTO fap_review_template_id_var
+      FROM templates 
+      WHERE name='default fap review template';
+
+    INSERT INTO questionaries(template_id, created_at, creator_id) VALUES (fap_review_template_id_var, NOW(), 1);
+
+    INSERT INTO topics(topic_title, is_enabled, sort_order, template_id) VALUES('New fap review', TRUE, 0, fap_review_template_id_var);
+
+    SELECT topics.topic_id
+    INTO fap_review_topic_id_var
+    FROM topics
+    WHERE topic_title='New fap review';
+
+    INSERT INTO templates_has_questions (question_id, template_id, topic_id, sort_order, config) VALUES('fap_review_basis', fap_review_template_id_var, fap_review_topic_id_var, 0, '{"required":false,"small_label":"","tooltip":""}');
+
+    SELECT questionaries.questionary_id
+    INTO questionary_id_var
+    FROM questionaries
+    WHERE template_id = fap_review_template_id_var;
+
+    UPDATE fap_reviews SET questionary_id = questionary_id_var;
+
+    UPDATE call set fap_review_template_id = fap_review_template_id_var;
```

Comment:
This will set the 'default fap review template' to all the calls in the database, which will affect the facilities running in production and they might not want to attach a fap review template. Shall we leave the column as `null` as it is and manually update the value in the UI, so as not to affect the exiting other facilities?

Reply:
I followed the strategy of proposal templates: the default FAP review template is the default 'dummy' template just like the default 'dummy' template for proposals. Since these templates are 'empty' without any question and the comment and grade fields are still part of the review table, this approach will not impact existing sites. The only impact will be the on the UI: instead of a single page there will be two steps, one for the entering the comment and grade and another one for the summary, but this is also pretty much the same for proposals.

Reply:
@yoganandaness pls let me know if you are fine with this response

Reply:
Sorry for the late response. 

1. As you can see in the screenshot, a proposal template is an important entity for a call, whereas a FAP review template should be able to be skipped(Correct me if i am wrong here). If my above thoughts are correct, then it is fine if we prepopulate the call with a Proposal template, whereas a FAP review template can be made empty and can be assigned directly in the UI, if needed. This way, facilities who don't have the need for a FAP review template can have this field as empty. 

<img width="1492" alt="Screenshot 2024-09-09 at 22 00 59" src="https://github.com/user-attachments/assets/9adc5823-fe23-4d9f-b346-160ad8874fe6">


2. I am not able to guess the two steps pages you were referring to. Would you be able to attach a screenshot or video?

Reply:
I guess that the FAP review is as important as the proposal (?), the same applies to their templates. 

In my code I followed the implementation strategy of proposal templates: I added a default FAP review template and assigned it to existing calls. 

Note that there is a default proposal template too that can be used for new calls if you don't want to create the template in advance or you are fine with the basic proposal configuration (title, abstract, PI, co-proposers)). The same applies to the FAP review templates: if you are fine with the comment + grade configuration, you can select the default FAP review templates. For existing calls the migration script will do this seamlessly. 

From point of view of implementation I think that it is better to have a default template than to add a lot of conditions and validations in the code to manage calls with no FAP review template.

Reply:
The new grading page has a review page (step):
![image](https://github.com/user-attachments/assets/5d550f82-6d55-4b1b-89c0-a98d13a708ab)

The old grading page was a single one:
![image](https://github.com/user-attachments/assets/dd664942-7ef7-4a34-8a6a-37d434c8ec1e)

Reply:
Thank you @gnyiri  for the detailed explanation. I am fine with your reasoning and had chat with @janosbabik as well. Sounds good to me. Sorry for holding this back. Approving it. Good work.